apps/client/src/app/pipes/pipes/quest-icon.pipe.ts

```diff
@@ -1,17 +1,38 @@
-import { Pipe, PipeTransform } from '@angular/core';
-import { LocalizedDataService } from '../../core/data/localized-data.service';
+import { ChangeDetectorRef, OnDestroy, Pipe, PipeTransform } from '@angular/core';
+import { of, Subject, Subscription } from 'rxjs';
+import { distinctUntilChanged, map, switchMap } from 'rxjs/operators';
+import { LocalizedLazyDataService } from '../../core/data/localized-lazy-data.service';
 
 @Pipe({
   name: 'questIcon',
-  pure: true
+  pure: false,
 })
-export class QuestIconPipe implements PipeTransform {
+export class QuestIconPipe implements PipeTransform, OnDestroy {
```

Comment:
I feel like some pipes have common code for everything related to subscriptions and all that, maybe creating a super class for them could be handy for future updates.

Reply:
Hm, I'll think on this one! It can be a little awkward for the same reason inheritance is awkward in components, when you override lifecycle methods it can be pretty bad if you forget to call the superclass', but there's probably a composition-based approach that shouldn't have that issue.